addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,338 @@
+    @api.depends('order_ids')
+    def _compute_currency_id(self):
+        self.currency_id = False
+        for wizard in self:
+            if wizard.count == 1:
+                wizard.currency_id = wizard.order_ids.currency_id
+
+    @api.depends('order_ids')
+    def _compute_company_id(self):
+        self.company_id = False
+        for wizard in self:
+            if wizard.count == 1:
+                wizard.company_id = wizard.order_ids.company_id
+
+    @api.depends('amount', 'fixed_amount', 'advance_payment_method', 'amount_to_invoice')
+    def _compute_display_invoice_amount_warning(self):
+        for wizard in self:
+            invoice_amount = wizard.fixed_amount
+            if wizard.advance_payment_method == 'percentage':
+                invoice_amount = wizard.amount / 100 * sum(wizard.order_ids.mapped('amount_total'))
+            wizard.display_invoice_amount_warning = invoice_amount > wizard.amount_to_invoice
+
+    @api.depends('order_ids')
+    def _compute_display_draft_invoice_warning(self):
+        for wizard in self:
+            wizard.display_draft_invoice_warning = wizard.order_ids.invoice_ids.filtered(lambda invoice: invoice.state == 'draft')
+
+    @api.depends('order_ids')
+    def _compute_invoice_amounts(self):
+        for wizard in self:
+            wizard.amount_invoiced = sum(wizard.order_ids._origin.mapped('amount_invoiced'))
+            wizard.amount_to_invoice = sum(wizard.order_ids._origin.mapped('amount_to_invoice'))
+
+    #=== ONCHANGE METHODS ===#
+
+    @api.onchange('advance_payment_method')
+    def _onchange_advance_payment_method(self):
+        if self.advance_payment_method == 'percentage':
+            amount = self.default_get(['amount']).get('amount')
+            return {'value': {'amount': amount}}
+
+    #=== CONSTRAINT METHODS ===#
+
+    def _check_amount_is_positive(self):
+        for wizard in self:
+            if (wizard.advance_payment_method == 'percentage' and wizard.amount <= 0.00) \
+                    or (wizard.advance_payment_method == 'fixed' and wizard.fixed_amount <= 0.00):
+                raise UserError(_('The value of the down payment amount must be positive.'))
+
+    # === ACTION METHODS ===#
+
+    def create_invoices(self):
+        self._check_amount_is_positive()
+        invoices = self._create_invoices(self.order_ids)
+        return self.order_ids.action_view_invoice(invoices=invoices)
+
+    def view_draft_invoices(self):
+        return {
+            'name': _('Draft Invoices'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'tree',
+            'views': [(False, 'list'), (False, 'form')],
+            'res_model': 'account.move',
+            'domain': [('state', '=', 'draft')], # To override
+        }
+
+    def _create_invoices(self, orders):
+        if self.advance_payment_method == 'delivered':
+            return orders._create_invoices(final=True, grouped=not self._invoice_grouped())
```

Comment:
Double check that we really want this `grouped=not self._invoice_grouped()`

Reply:
It looks weird indeed (but is correct). I changed the name to consolidated billing to make it less confusing.